tools/pybind11Tools.cmake

```diff
@@ -51,10 +51,12 @@ endfunction()
 macro(_pybind11_return_if_cxx_and_linker_flags_work feature cxxflags linkerflags cxx_out linker_out)
   _pybind11_check_cxx_and_linker_flags(${cxxflags} ${linkerflags} check_success)
   if(check_success)
-    separate_arguments(cxxflags)
-    separate_arguments(linkerflags)
-    set(${cxx_out} ${cxxflags} PARENT_SCOPE)
-    set(${linker_out} ${linkerflags} PARENT_SCOPE)
+    set(cxx ${cxxflags})
+    set(linker ${linkerflags})
+    separate_arguments(cxx)
+    separate_arguments(linker)
```

Comment:
`separate_arguments(linkerflags)` didn't work (silently) because this is a macro with a string literal, and so linkerflags apparently isn't a proper variable.

The space-versus-semicolon is a nuisance imposed by cmake: we need space-separated flags when passing multiple flags to check_cxx_compiler_flag (both in the direct argument and in the overridden `CMAKE_REQUIRED_LIBRARIES` variable), but need semicolon-separated when passing the flags to `target_compile_options()`/`target_link_libraries()`.  Otherwise the latter quotes it and passes it as a single argument `"-flto -fno-fat-lto-objects"`, which of course fails.